community/modules/compute/schedmd-slurm-gcp-v6-nodeset/main.tf

```diff
@@ -175,3 +175,21 @@ data "google_compute_reservation" "reservation" {
     # Add a validation that if reservation.project != var.project_id it should be a shared reservation
   }
 }
+
+data "google_compute_machine_types" "machine_types_by_zone" {
+  for_each = local.zones
+  filter   = format("name = \"%s\"", local.nodeset.machine_type)
+  zone     = each.value
+}
+
+locals {
+  machine_types_by_zone      = data.google_compute_machine_types.machine_types_by_zone
+  zones_without_machine_type = [for k, v in local.machine_types_by_zone : k if length(v.machine_types) == 0]
+}
+
+check "is_machine_type_available_in_zones" {
```

Comment:
1. You have the [wrong syntax for checks](https://developer.hashicorp.com/terraform/language/checks#syntax). The nice thing about checks is that data sources are "scoped" inside the check block which allows them to fail.

2. I think you are using the wrong condition. The assertion we want is "at least 1 zone has the machine type" not "there are 0 zones without the machine type". Both your examples should succeed.

3. Despite (1), it's interesting to me that the condition is failing in the check block rather than the data source. I would think that it would fail at the data source stage. If the google_compute_machine_types data source can successfully return the empty list, it would actually be more desirable to **fail** as a precondition than to warn with a check.

Can you please verify the behavior of the data source on a to confirm that it can return empty without failing?

e.g.

```hcl
data "google_compute_machine_types" "machine_types_by_zone" {
  for_each = toset(["us-central1-b", "us-central1-c"])
  filter   = "name = \"h3-standard-88\""
  zone     = each.value
}
```

Run `terraform state list` and `terraform state show data.google_compute_machine_types.machine_types_by_zone[\"us-central1-b\"]`. If this returns the empty list after `terraform apply`, then we are safe to fail on the data source instead of warning in a check block. So long as you fix the boolean logic (2).

Reply:
Verified that 3 works. The data source does not fail.

Updated the code to use `precondition` with `terraform_data` resource as discussed over the chat. If none of the zones have the machine type available, Terraform will throw an error and exit.

Tested the code on success and failure scenarios (updated the PR description).